sdk/core/Azure.Core/src/DynamicData/DynamicData.cs

```diff
@@ -476,7 +476,7 @@ public override int GetHashCode()
         private string DebuggerDisplay => _element.DebuggerDisplay;
 
 #if !NET5_0
-        [RequiresUnreferencedCode("This class utilizes reflection-based JSON serialization and deserialization which is not compatible with trimming.")]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", Justification = "This JsonConverter is used with the DynamicData class which is already marked as RequiresUnreferencedCode.")]
```

Comment:
I don't really agree with this change. Adding `RequiresUnreferencedCode` here is the right thing to do. Since it's only used from `DynamicData` any usage should not produce new warnings. Suppressions are dangerous... basically always. If at all possible, try to avoid using them. In this case the `RequiresUnreferencedCode` is the same cost and is the cleaner solution.

(Same comment for the other cases of this below).

Reply:
I should probably clarify one thing. The `RequiresUnreferencedCode` is intentionally designed such that it doesn't propagate to nested types:

```C#
[RequiresUnreferencedCode("")]
class Parent
{
    public void Method()
    {
        // Any warning here is suppressed by the presence of RequiresUnreferencedCode on the type         
        DoSomethingIncompatible();
    }
    
    class Nested
    {
        public void NestedMethod()
        {
            // IL#### - this will produce a warning, since the attribute doesn't propagate to nested types
            DoSomethingIncompatible();
        }
    }
}
```

That's why adding the attribute onto the methods of the class is not necessary, since it does propagate to methods. But we should add it to nested types which need it still.

This behavior is intentional, I can explain the reasons for that decision, but it's probably not very interesting for this discussion.

Reply:
@vitek-karas I did originally have it that way, but the JsonConverters are used by an attribute on the class or struct. Even if the class is annotated with RequiresUnreferencedCode, I still get warnings from the attribute (no matter what order the attributes are in). I wasn't sure how I could annotate an annotation?

Reply:
Sorry - I missed that part... Hmm that's unfortunate.

It is intentional that the `RequiresUnreferencedCode` on the type doesn't suppress warnings from attributes on the type (again, complicated reasons). But that just explains why you couldn't get it to work.

In that case this is probably OK - I think it's worth a comment that they're used the attributes on the parent type.

Reply:
Added a comment / clarified the message a bit.